### ISSUE TITLE
allow to use Spring property placeholder for cache-deserialized-values map config

### ISSUE THREAD

**Issue by dsukhoroslov** [hazelcast/hazelcast#8487]
Currently the system throws exception if I try to use property placeholder for this attribute.

the attribute specified as:

```
<xs:attribute name="cache-deserialized-values" use="optional" type="cache-deserialized"/>

    <xs:simpleType name="cache-deserialized">
        <xs:restriction base="non-space-string">
            <xs:enumeration value="NEVER"/>
            <xs:enumeration value="ALWAYS"/>
            <xs:enumeration value="INDEX-ONLY"/>
        </xs:restriction>
    </xs:simpleType>
```

it'd be better to specify the type as

```
    <xs:simpleType name="cache-deserialized-type">
        <xs:union memberTypes="cache-deserialized non-space-string"/>
    </xs:simpleType>
```

**Comment by dsukhoroslov** [comment]
with the new `parameterizedValueType `introduced in hazelcast-string-3.8.xsd the attribute definition should be:

`<xs:attribute name="cache-deserialized-values" use="optional" type="parameterized-cache-deserialized"/>
`
```
<xs:simpleType name="parameterized-cache-deserialized">
    <xs:union memberTypes="cache-deserialized parameterizedValueType"/>
</xs:simpleType>
```
any chance to add it in 3.8?

**Comment by emrahkocaman** [comment]
Thanks for reporting @dsukhoroslov . It didn't make it to 3.8 unfortunately, but will be available with 3.9.
